xds/internal/xdsclient/tests/eds_watchers_test.go

```diff
@@ -52,6 +52,31 @@ const (
 	edsPort3 = 3
 )
 
+type endpointsUpdateErrTuple struct {
+	update xdsresource.EndpointsUpdate
+	err    error
+}
+
+type endpointsWatcher struct {
+	updateCh *testutils.Channel
+}
+
+func newEndpointsWatcher() *endpointsWatcher {
+	return &endpointsWatcher{updateCh: testutils.NewChannel()}
+}
+
+func (ew *endpointsWatcher) OnUpdate(update *xdsresource.EndpointsResourceData) {
+	ew.updateCh.Send(endpointsUpdateErrTuple{update: update.Resource})
+}
+
+func (ew *endpointsWatcher) OnError(err error) {
+	ew.updateCh.SendOrFail(endpointsUpdateErrTuple{err: err})
+}
+
+func (ew *endpointsWatcher) OnResourceDoesNotExist() {
+	ew.updateCh.SendOrFail(endpointsUpdateErrTuple{err: xdsresource.NewErrorf(xdsresource.ErrorTypeResourceNotFound, "Endpoints not found in received response")})
```

Comment:
This error string corresponds to the one used in `xds/internal/xdsclient/tests/resource_update_test.go`. Earlier, since the code was calling the now deleted `xdsclient.WatchEndpoints` method implemented in `xds/internal/xdsclient/clientimpl_watchers.go`, and where we had a `endpointsWatcher` implementation that returned the following error upon resource-not-found.

```
func (c *endpointsWatcher) OnResourceDoesNotExist() {
	err := xdsresource.NewErrorf(xdsresource.ErrorTypeResourceNotFound, "resource name %q of type Endpoints not found in received response", c.resourceName)
	c.cb(xdsresource.EndpointsUpdate{}, err)
}
```

Now, if I change the error string here, I would have to touch the tests as well, which I feel is unnecessary. Let me know what you think.

Reply:
I see the thing you refactored from going through: https://github.com/grpc/grpc-go/pull/6414/files#diff-23960407248b82145ea16f839405e2a47798a0de947a1f3c72aacbc36769fa13L132. And since the xDS Client would call this function in the case where a resource doesn't exist, I just wanted a string saying "a Resource of type Endpoints" rather than just "Endpoints", because it's something that isn't present that is a type, corresponding to what you changed from master. I don't feel too strongly about this though. I was just nitpicking about "Endpoints not found in received response".